graph.go

```diff
@@ -98,21 +98,26 @@ func (g *Graph) Commit() (*QueryResult, error) {
 }
 
 // Query executes a query against the graph.
-func (g *Graph) Query(q string) (*QueryResult, error) {
-	
-	r, err := g.Conn.Do("GRAPH.QUERY", g.Id, q, "--compact")
+func (g *Graph) Query(q string, timeout ...int)(*QueryResult, error) {
```

Comment:
@jeffreylovitz we can't change the function signature without breaking the customer app. I suggest two approaches:
- we add two new methods `QueryWithTimeout` and `ParameterizedQueryWithTimeout` 
- we add two new methods `QueryWithOptions` and `ParameterizedQueryWithOptions` and create a new QueryOptions struct to include timeout ( and other future features ).

Reply:
@filipecosta90 I'm resurrecting this PR as it resolves a customer bug.

The second option you listed sounds like the most robust design, so I've implemented it here. Let me know if it needs work!